variants/github_actions_ci/workflows/deploy_to_ec2.yml

```
@@ -0,0 +1,75 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+      slack_icon:
+        type: string
+        required: false
+        default: https://github.com/ackama.png?size=48
+      slack_channel:
+        type: string
+        required: false
+      slack_username:
+        type: string
+        required: false
+        default: "Github Actions CI"
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: false
+jobs:
+  deploy_to_ec2_via_oidc:
+    name: Deploy to AWS EC2 with Capistrano, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    permissions:
+      # These permissions are needed to interact with GitHub's OIDC Token
+      # endpoint.
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ inputs.assume_role_arn }}
+          aws-region: ${{ inputs.aws_region }}
+      - name: Install SSH key for production
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.ssh_private_key }}
+          known_hosts: unnecessary
+      - name: Deploy application
+        run: bundle exec cap ${{ inputs.environment }} deploy
+      - name: Report to Slack
+        # 2022-09-16: Github Actions cannot currently reference `secrets` in an
+        # if expression. Workaround is to copy to secret to env var. In this
+        # case we already need the secret in an env var. Details:
+        # https://github.com/actions/runner/issues/520
+        if: "${{ env.SLACK_CHANNEL != '' }}"
```

🐘 we should be able to use `inputs` right? I think that'd be more consistent:

```suggestion
        if: "${{ inputs.SLACK_CHANNEL != '' }}"
```